src/dsp_tools/commands/xmlupload/models/ingest.py

```diff
@@ -84,3 +115,40 @@ def ingest(self, shortcode: str, filepath: Path) -> IngestResponse:
                     raise PermanentConnectionError(log_msg)
             except requests.exceptions.RequestException as e:
                 raise PermanentConnectionError(f"{err}. {e}")
+
+    def get_bitstream_info(
+        self,
+        bitstream: XMLBitstream,
+        permissions_lookup: dict[str, Permissions],
+        res_label: str,
+        res_id: str,
+    ) -> tuple[bool, None | BitstreamInfo]:
+        """Uploads a file to the ingest server and returns the BitstreamInfo."""
+        try:
+            res = self._ingest(Path(self.imgdir) / Path(bitstream.value))
+            msg = f"Uploaded file '{bitstream.value}'"
+            print(f"{datetime.now()}: {msg}")
+            logger.info(msg)
+            permissions = permissions_lookup.get(bitstream.permissions) if bitstream.permissions else None
+            return True, BitstreamInfo(bitstream.value, res.internal_filename, permissions)
+        except PermanentConnectionError as err:
+            msg = f"Unable to upload file '{bitstream.value}' " f"of resource '{res_label}' ({res_id})"
+            print(f"{datetime.now()}: WARNING: {msg}: {err.message}")
+            logger.opt(exception=True).warning(msg)
+            return False, None
+
+
+@dataclass(frozen=True)
+class BulkIngestedAssetClient(AssetClient):
+    """Client for handling media info, if the assets were bulk ingested previously."""
+
+    def get_bitstream_info(
+        self,
+        bitstream: XMLBitstream,
+        permissions_lookup: dict[str, Permissions],
+        res_label: str,  # noqa: ARG002
+        res_id: str,  # noqa: ARG002
+    ) -> tuple[bool, BitstreamInfo | None]:
+        """Uploads a file to the ingest server and returns the BitstreamInfo."""
```

Comment:
copy-paste error

Reply:
actually, I think this was an AI hallucination... thanks for spotting!